client/src/components/profile/components/TimeLine.js

```diff
@@ -39,20 +60,26 @@ class TimelineInner extends Component {
 
     this.state = {
       solutionToView: null,
-      solutionOpen: false
+      solutionOpen: false,
+      pageNo: 1,
+      totalPages: Math.ceil(props.sortedTimeline.length / ITEMS_PER_PAGE)
```

Comment:
This is the only thing I'm not 100% on.  I think it would be better to use state to hold, well, just the current state of the page.  `totalPages` does not change unless the props change, so it would be a bit better to pass in `totalPages` to `TimelineInner` as a separate prop.  That way there's still a single source of truth and state is reserved for recording user actions.

Possibly I'm being overly restrictive about how state can be used, though.

I'm probably being overly pedantic, though, so

Reply:
Thanks for the suggestion. I have made totalPages as a prop instead of state.